docker-micro-benchmark/helpers/errors.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helpers
+
+import (
+	"fmt"
+	"sync"
+)
+
+const errorTopNum = 5
+
+// Error is the error returned by functions in docker_helpers
+type Error struct {
+	sync.RWMutex
+	errs  map[string][]error
+	total map[string]int
+}
+
+func newError() *Error {
+	return &Error{
+		errs:  make(map[string][]error),
+		total: make(map[string]int),
+	}
+}
+
+func (e *Error) isNil() bool {
+	e.RLock()
+	defer e.RUnlock()
+	return len(e.errs) <= 0
+}
+
+func (e *Error) add(l string, err error) {
+	e.Lock()
+	defer e.Unlock()
+	e.total[l] = e.total[l] + 1
+	if err != nil {
+		e.errs[l] = append(e.errs[l], err)
```

Comment:
I am slightly wary of keeping all the error messages in memory. Maybe we need a flag t turn off this. Let's add a TODO for this.